src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -20,7 +20,7 @@ public sealed partial class JsonSerializerOptions
         /// <remarks>
         /// Once serialization or deserialization occurs, the list cannot be modified.
         /// </remarks>
-        public IList<JsonConverter> Converters => _converters;
+        public IList<JsonConverter> Converters => _converters ??= new ConverterList(this);
```

Comment:
This isn't thread-safe, in that two threads concurrently accessing this could end up getting different ConverterList instances.  Is that ok?

And, it seems like this could transition from null to non-null even after being marked as isreadonly.  Is that ok?

Reply:
It's ok for both cases, all logic consuming that field (including equality comparison in the global cache) has been updated so that `null` instances are equated to empty instances.

Reply:
It can, of course, result in issues when multiple threads attempt to configure the same mutable instance, but generally speaking the type is not designed to be thread safe while mutable.